src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -452,6 +452,15 @@ private bool TryClose()
 
                 _context.StreamLifetimeHandler.OnStreamConnectionError(ex);
             }
+            catch (ConnectionResetException ex)
+            {
+                // TODO: This is temporary. Don't want to tie HTTP/3 layer to one transport.
+                // This is here to check what other exceptions can cause ConnectionResetException.
+                Debug.Assert(ex.InnerException is QuicStreamAbortedException);
```

Comment:
This really only affects local development, right? I'm not sure this is worth checking in especially if we're not confident in it. Maybe just keep it as a stash you reapply locally? I guess it's not a big deal either way.

Reply:
Yes.

I want to check because I'm not confident and I want it to break locally